### PR TITLE
refactor(samples): add meetings sample authentication

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Pull Request

## Description

The changes in this pull request are scoped around implementing the `Authentication` section of the new Meetings Sample.

Fixes # [SPARK-143728](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-143728)

### CODE REVIEW

**Comment by taymoork2** [comment]
Change body css to include `font-family: sans-serif`

File: packages/node_modules/samples/browser-meetings/index.html [hunk 1]
```diff
@@ -34,13 +38,24 @@ <h2>Meetings</h2>
...
+      <p>This section contains all of the logic needed to fully authenticate and prepare the Webex JS SDK for meetings usage. There are two steps that need to be performed before the meetings functionality of the JS SDK is available for use:</p>
+      <ul>
+        <li>Webex.init() => Initialize the Webex JS SDK</li>
+        <li>webex.meetings.register() => Register the local device using</li>
+      </ul>
+      <p>Once the Webex JS SDK instance is ready and registered, The rest of the meetings functionality of the Webex JS SDK becomes available. These values can be confirmed by checking:</p>
+      <ul>
+        <li>webex.ready => Determine if the webex instance is ready</li>
+        <li>webex.meetings.registered => Determine if the webex instances local device is registered</li>
+      </ul>
```
Comment (taymoork2) [inline packages/node_modules/samples/browser-meetings/index.html:51]: This is too wordy and isn't in the order of expected execution
This can be a bit troublesome and confusion, however the explanation itself is needed to explain what the api does

I suggest adding simple collapsible add the end of [each] section that has a simple example on how to implement, like how a lot of CSS frameworks have in their docs

```js
let webex = window.webex = Webex.init({
    credentials: {
      access_token: tokenElm.value
    }
  });

  webex.once('ready', () => {
    webex.meetings.register()
    .then(() => {
      console.log('Authentication#register() :: successfully registered');
    })
    .catch((error) => {
      console.warn('Authentication#register() :: error registering', error);
    })
  });
```

File: packages/node_modules/samples/browser-meetings/app.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+const tokenElm = document.querySelector('#access-token');
+const saveElm = document.querySelector('#access-token-save');
+const authStatusElm = document.querySelector('#access-token-status');
+const registerElm = document.querySelector('#registration-register');
+const unregisterElm = document.querySelector('#registration-unregister');
+const registrationStatusElm = document.querySelector('#registration-status');
+
+function saveCredentials() {
+  console.log('Authentication#saveCredentials()');
```
Comment (adamweeks) [inline packages/node_modules/samples/browser-meetings/app.js:41]: I'm worried that we're creating too much of an abstraction here. `saveCredentials` isn't an actual function a user could call and might cause confusion. Would prefer this to directly refer to the `Webex.init` function we call.

File: packages/node_modules/samples/browser-meetings/app.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    console.log('Authentication#saveCredentials() :: Webex Ready');
+    registerElm.disabled = false;
+    authStatusElm.innerHTML = 'Saved';
+  });
+}
+
+function register() {
+  console.log('Authentication#register()');
+  registerElm.disabled = true;
+  registrationStatusElm.innerHTML = 'Registering...';
```
Comment (adamweeks) [inline packages/node_modules/samples/browser-meetings/app.js:63]: Let's add a `unregisterElm.disabled = true;` here too so we don't attempt to unregister during our register.

File: packages/node_modules/samples/browser-meetings/index.html [hunk 4]
```diff
@@ -29,18 +54,24 @@
     <h1>THIS IS A WORK IN PROGRESS SAMPLE</h1>
     <section>
       <h2>Meetings</h2>
-      <p>[[APPLICATION DESCRIPTION]]</p>
+      <p>This is the meetings plugin sample for the Webex JS SDK.</p>
+      <p>Each of the following sections within this sample contain forms that perform specific actions within the meetings plugin of the Webex JS SDK.</p>
+      <p>To find the exact logic performed by each of the form input elements, please inspect the browser's developer console, as it will provide a reference to a function in this application that performs the selected action within the ./app.js file.</p>
     </section>
 
     <section>
```
Comment (taymoork2) [inline packages/node_modules/samples/browser-meetings/index.html:62]: This is what I'm suggesting each section should look like
It's pretty much what 90% of CSS framework docs are setup as

```html
<section>
  <h2>Authentication</h2>
  <p>This section contains all of the logic needed to fully authenticate and prepare the Webex JS SDK for meetings usage. Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token, and once ready, registered as a device.</p>
  <p>There are two steps that need to be performed before the meetings functionality of the JS SDK is available for use are:</p>
  <ul>
  	<li>Webex.init() =&gt; Initialize the Webex JS SDK</li>
    <li>webex.meetings.register() =&gt; Register the local device using</li>
  </ul>
  <p>Once the Webex JS SDK instance is ready and registered, The rest of the meetings functionality of the Webex JS SDK becomes available. These values can be confirmed by checking:</p>
  <ul>
    <li>webex.ready =&gt; Determine if the webex instance is ready</li>
    <li>webex.meetings.registered =&gt; Determine if the webex instances local device is registered</li>
  <form id="credentials">
    <fieldset>
      <legend>Credentials</legend>
      <input id="access-token" name="accessToken" placeholder="Access Token" value="" type="text">
      <button id="access-token-save" type="button" onclick="saveCredentials()">Save</button>
      <label id="access-token-status"></label>
    </fieldset>
  </form>
  <br>
  <form lpformnum="1">
    <fieldset>
      <legend>Registration</legend>
      <button id="registration-register" type="button" onclick="register()" disabled="">Register</button>
      <button id="registration-unregister" type="button" onclick="unregister()" disabled="">Unregister</button>
      <label id="registration-status">Not Registered</label>
    </fieldset>
  </form>
  <div class="collapsible" data-collapsed="false">Code Sample</div>
    <div class="collapsible-content" data-collapsed="false">
    <code wrap="false">
      let webex = (window.webex = Webex.init({
        credentials: {
          access_token: <YOUR ACCESS TOKEN HERE>,
        },
      }));

      webex.once('ready', () => {
        webex.meetings.register()
          .then(() => {
            console.log('Authentication#register() :: successfully registered');
          })
          .catch((error) => {
            console.warn('Authentication#register() :: error registering', error);
          });
      });
	</code>        
  </div>
</section>
```

File: packages/node_modules/samples/browser-meetings/app.js (hunk 3 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:63]: Don't need to do this, there is no way that the `unregister` button can be enabled without a registered meetings plugin. Let me know if you found a case that it is.

Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:63]: Implemented anyways!

File: packages/node_modules/samples/browser-meetings/app.js (hunk 2 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:41]: Updated to `initWebex()`

File: packages/node_modules/samples/browser-meetings/app.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+// Authentication Section
+
+const tokenElm = document.querySelector('#access-token');
+const saveElm = document.querySelector('#access-token-save');
+const authStatusElm = document.querySelector('#access-token-status');
+const registerElm = document.querySelector('#registration-register');
+const unregisterElm = document.querySelector('#registration-unregister');
+const registrationStatusElm = document.querySelector('#registration-status');
+
+function saveCredentials() {
```
[the remark this answers is not in the document]
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:40]: went with `initWebex()`

File: packages/node_modules/samples/browser-meetings/index.html (hunk 1 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html]: Cleaning this up and leaving it as is for now, a later PR will add this feature/documentation.

File: packages/node_modules/samples/browser-meetings/index.html (hunk 4 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html:62]: Leaving this as is for now, we will be adding features as described in this comment in a later PR.

File: packages/node_modules/samples/browser-meetings/index.html [hunk 6]
```diff
@@ -29,18 +54,24 @@
     <h1>THIS IS A WORK IN PROGRESS SAMPLE</h1>
     <section>
       <h2>Meetings</h2>
-      <p>[[APPLICATION DESCRIPTION]]</p>
+      <p>This is the meetings plugin sample for the Webex JS SDK.</p>
+      <p>Each of the following sections within this sample contain forms that perform specific actions within the meetings plugin of the Webex JS SDK.</p>
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/index.html:58]:
```suggestion
      <p>Each of the following sections within this sample perform specific actions that map to methods in the meetings plugin of the Webex JS SDK.</p>
```

File: packages/node_modules/samples/browser-meetings/index.html [hunk 7]
```diff
@@ -29,18 +54,24 @@
...
+      <p>To find the exact logic performed by each of the form input elements, please inspect the browser's developer console, as it will provide a reference to a function in this application that performs the selected action within the ./app.js file.</p>
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/index.html:59]:
```suggestion
      <p>To find the exact logic performed by an action, please inspect the browser's developer console. We log in the console functions in this application that performs the selected action within the <code>./app.js</code> file.</p>
```

File: packages/node_modules/samples/browser-meetings/index.html [hunk 8]
```diff
@@ -29,18 +54,24 @@
...
     </section>
 
     <section>
       <h2>Authentication</h2>
-      <p>[[AUTHENTICATION DESCRIPTION]]</p>
+      <div class="collapsible" data-collapsed="true">Details</div>
+      <div class="collapsible-content" data-collapsed="true">
+        <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token, and once ready, registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/index.html:66]:
```suggestion
<p>
  Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token.
  You may get test access token from <a target="_blank">https://developer.webex.com/docs/api/getting-started</a>. 
  Once the JS SDK is has authenticated your token, it then must be registered as a device.
  Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.
</p>
```

File: packages/node_modules/samples/browser-meetings/app.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  this.nextElementSibling.setAttribute('data-collapsed', newState);
+}
+
+collapsibles.forEach(
+  (collapsible) => {
+    collapsible.addEventListener('click', collapse);
+  }
+);
+
+// Authentication Section
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/app.js:31]: I like longer comment delineators, I find it makes it easier to glance over sections
```suggestion
// Authentication Section --------------------------------------------------
```

File: packages/node_modules/samples/browser-meetings/app.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+/* eslint-disable require-jsdoc */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-console */
+/* eslint-disable no-global-assign */
+/* eslint-disable no-multi-assign */
+
+// Globals
+let webex;
+
+// Collapsible Setup
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/app.js:14]: I like longer comment delineators, I find it makes it easier to glance at sections
```suggestion
// Collapsible Setup --------------------------------------------------
```

File: packages/node_modules/samples/browser-meetings/index.html (hunk 6 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html:58]: implemented.

File: packages/node_modules/samples/browser-meetings/index.html (hunk 7 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html:59]: implemented.

File: packages/node_modules/samples/browser-meetings/index.html (hunk 8 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html:66]: Implemented, but fixed the `<a />` element, it was missing the `href` attribute, see the most up-to-date changes.

File: packages/node_modules/samples/browser-meetings/app.js (hunk 9 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:31]: Implemented.

File: packages/node_modules/samples/browser-meetings/app.js (hunk 10 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js:14]: Implemented.

File: packages/node_modules/samples/browser-meetings/app.js [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+  this.setAttribute('data-collapsed', newState);
+  this.nextElementSibling.setAttribute('data-collapsed', newState);
+}
+
+collapsibles.forEach(
+  (collapsible) => {
+    collapsible.addEventListener('click', collapse);
+  }
+);
```
Comment (taymoork2) [inline packages/node_modules/samples/browser-meetings/app.js:29]: Move this code to a `<script>` in the `<head>`
This code is getting blocked from executing due to `webex.min.js` waiting to download

File: packages/node_modules/samples/browser-meetings/index.html [hunk 12]
```diff
@@ -29,18 +54,24 @@
...
+      <p>To find the exact logic performed by an action, please inspect the browser's developer console. We log in the console functions in this application that performs the selected action within the <code>./app.js</code> file.</p>
     </section>
 
     <section>
       <h2>Authentication</h2>
-      <p>[[AUTHENTICATION DESCRIPTION]]</p>
+      <div class="collapsible" data-collapsed="true">Details</div>
+      <div class="collapsible-content" data-collapsed="true">
+        <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token. You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>. Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
+      </div>
```
Comment (taymoork2) [inline packages/node_modules/samples/browser-meetings/index.html:67]: I personally didn't want collapsable’s on descriptions, only on code samples which got veto'd
```suggestion
      <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token. You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>. Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
```

File: packages/node_modules/samples/browser-meetings/app.js (hunk 11 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/app.js]: Removing the collapsable, as that would require two separate files or an on-ready event handler.

File: packages/node_modules/samples/browser-meetings/index.html (hunk 12 above)
Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html]: Removed the collapsable.

File: packages/node_modules/samples/browser-meetings/index.html (hunk 8 above)
Comment (lalli-flores) [reply packages/node_modules/samples/browser-meetings/index.html]: yeah, just meant to quickly show you what I meant 😅

File: packages/node_modules/samples/browser-meetings/index.html [hunk 13]
```diff
@@ -29,18 +34,21 @@
...
-      <p>[[APPLICATION DESCRIPTION]]</p>
+      <p>This is the meetings plugin sample for the Webex JS SDK.</p>
+      <p>Each of the following sections within this sample perform specific actions that map to methods in the meetings plugin of the Webex JS SDK.</p>
+      <p>To find the exact logic performed by an action, please inspect the browser's developer console. We log in the console functions in this application that performs the selected action within the <code>./app.js</code> file.</p>
     </section>
 
     <section>
       <h2>Authentication</h2>
-      <p>[[AUTHENTICATION DESCRIPTION]]</p>
+      <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token. You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>. Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
```
Comment (taymoork2) [inline packages/node_modules/samples/browser-meetings/index.html:44]: Make it slightly more readable
```suggestion
      <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token.
      <p>You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>.</p>
      <p>Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
```

Comment (InteractiveTimmy) [reply packages/node_modules/samples/browser-meetings/index.html]: implemented.

File: packages/node_modules/samples/browser-meetings/index.html [hunk 14]
```diff
@@ -29,18 +34,23 @@
     <h1>THIS IS A WORK IN PROGRESS SAMPLE</h1>
     <section>
       <h2>Meetings</h2>
-      <p>[[APPLICATION DESCRIPTION]]</p>
+      <p>This is the meetings plugin sample for the Webex JS SDK.</p>
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/index.html:37]: My previous comment meant to have each sentence on its own line, but not necessarily in each own `<p>` tag. This way is the paragraph is more readable from the source but still looks like one paragraph when the page displays.

File: packages/node_modules/samples/browser-meetings/index.html [hunk 15]
```diff
@@ -29,18 +34,23 @@
...
+      <p>Each of the following sections within this sample perform specific actions that map to methods in the meetings plugin of the Webex JS SDK.</p>
+      <p>To find the exact logic performed by an action, please inspect the browser's developer console. We log in the console functions in this application that performs the selected action within the <code>./app.js</code> file.</p>
     </section>
 
     <section>
       <h2>Authentication</h2>
-      <p>[[AUTHENTICATION DESCRIPTION]]</p>
+      <p>Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token.</p>
```
Comment (lalli-flores) [inline packages/node_modules/samples/browser-meetings/index.html:44]: My previous comment meant to have each sentence on its own line, but not necessarily in each own `<p>` tag. This way is the paragraph is more readable from the source but still looks like one paragraph when the page displays.

File: packages/node_modules/samples/browser-meetings/index.html [hunk 16]
```diff
@@ -29,18 +34,23 @@
...
+      <p>You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>.</p>
+      <p>Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.</p>
+
       <form id="credentials">
         <fieldset>
           <legend>Credentials</legend>
           <input id="access-token" name="accessToken" placeholder="Access Token" value="" type="text" />
-          <button id="access-token-save" type="button">Save</button>
-          <label id="access-token-status">[[CREDENTIALS STATUS]]</label>
+          <button id="access-token-save" type="button" onclick="initWebex()">Webex.init()</button>
```
Comment (adamweeks) [inline packages/node_modules/samples/browser-meetings/index.html:52]: Really like using `onclick` to map with the app.js code here. 👍

File: packages/node_modules/samples/browser-meetings/index.html (hunk 15 above)
Comment (taymoork2) [reply packages/node_modules/samples/browser-meetings/index.html:44]: I asked to new line everything, since it just looked like a soup of text as a paragraph

Comment (lalli-flores) [reply packages/node_modules/samples/browser-meetings/index.html:44]: yeah, that's what I asked for too, but they don't need to be on separate `<p>` tags

Comment (taymoork2) [reply packages/node_modules/samples/browser-meetings/index.html:44]: Can you leave a suggestion for Tim to easily commit the changes?

Comment (lalli-flores) [reply packages/node_modules/samples/browser-meetings/index.html:44]: My suggestion:
```suggestion
<p>
  Before using the meetings functionality of the Webex JS SDK, the Webex JS SDK must be initialized using a valid token.
  You may get test access token from our developer portal: <a href="https://developer.webex.com/docs/api/getting-started" target="_blank">https://developer.webex.com/docs/api/getting-started</a>.
  Once the JS SDK is has authenticated your token, it then must be registered as a device. Once these two steps have been performed, the meetings functionality of the Webex JS SDK should be available for usage.
</p>
```
